compiler+runtime/src/cpp/jank/runtime/obj/inst.cpp

```diff
@@ -0,0 +1,84 @@
+#include <chrono>
+#include <iostream>
+#include <locale>
+#include <sstream>
+
+#include <jank/runtime/core/make_box.hpp>
+#include <jank/runtime/obj/inst.hpp>
+#include <jank/runtime/rtti.hpp>
+#include <jank/util/fmt/print.hpp>
+
+namespace jank::runtime::obj
+{
+  static jtl::ref<inst_time_point> now()
+  {
+    return jtl::make_ref<inst_time_point>(std::chrono::system_clock::now());
+  }
+
+  inst::inst()
+    : value{ now() }
+  {
+  }
+
+  static jtl::ref<inst_time_point> from_string(jtl::immutable_string const &s)
+  {
+    inst_time_point o;
+    std::istringstream is{ static_cast<std::string>(s) };
+    is.imbue(std::locale("en_US.utf-8"));
+    is >> std::chrono::parse("%FT%T", o);
```

Comment:
I'm going to update this bit to loop through various supported formats in hopes of finding one that succeeds.

Reply:
Does Clojure use different `#inst` formats?

Reply:
I think for printing, it's one format but for parsing, it supports many. The following are all valid:
```clojure
#inst "2025-09-01T19:10:28.691-00:00"
#inst "2025-09-01T19:10:28.691"
#inst "2025-09-01T19:10:28"
#inst "2025-09-01"
```